File: plugins/clock/clock.1s.py
#!/usr/bin/env python3
"""Clock Plugin - Shows current time using Crossbar API"""
import subprocess
from datetime import datetime

def crossbar(*args):
    """Execute crossbar command and return output"""
    try:
        result = subprocess.run(['crossbar'] + list(args), capture_output=True, text=True, timeout=5)
        return result.stdout.strip() if result.returncode == 0 else None
    except Exception:
        return None

# Get time from Crossbar API or fallback to Python
time_str = crossbar('time') or datetime.now().strftime('%H:%M:%S')
date_str = crossbar('date') or datetime.now().strftime('%Y-%m-%d')

print(f"🕐 {time_str}")
print("---")
print(f"Time: {time_str}")
print(f"Date: {date_str}")
print("---")
print("Refresh | refresh=true")

File: plugins/clock/clock.1s.lua
-- clock.1s.lua
-- Simple clock plugin using embedded Lua interpreter

local function env(name, default)
    local value = crossbar.env(name, default)
    if value == nil or value == '' then
        return default
    end
    return value
end

local format = env('CLOCK_FORMAT', 'HH:mm:ss')
local time = crossbar.time(format)
print('🕐 ' .. time)
